front/temporal/data_retention/activities.ts

```diff
@@ -69,29 +70,29 @@ export async function purgeConversationsBatchActivity({
 
     const auth = await Authenticator.internalAdminForWorkspace(workspace.sId);
 
-    let conversations: ConversationResource[];
-    let nbConversationsDeleted = 0;
+    const conversations = await ConversationResource.listAllBeforeDate(auth, {
+      batchSize: WORKSPACE_CONVERSATIONS_BATCH_SIZE,
+      cutoffDate,
+      includeDeleted: true,
+      includeTest: true,
+    });
 
-    do {
-      conversations = await ConversationResource.listAllBeforeDate(auth, {
-        batchSize: WORKSPACE_CONVERSATIONS_BATCH_SIZE,
+    logger.info(
+      {
+        workspaceId,
+        retentionDays,
         cutoffDate,
-        includeDeleted: true,
-        includeTest: true,
-      });
+        nbConversations: conversations.length,
+      },
+      "Purging conversations for workspace."
+    );
 
-      logger.info(
-        {
-          workspaceId,
-          retentionDays,
-          cutoffDate,
-          nbConversations: conversations.length,
-        },
-        "Purging conversations for workspace."
-      );
+    // Process deletions in batches to heartbeat regularly.
+    for (let i = 0; i < conversations.length; i += HEARTBEAT_INTERVAL) {
```

Comment:
Previous batch do/while was useless, we returned the whole conversations at first call.